lib/setupmanagers/hckclient.rb

```diff
@@ -18,36 +18,42 @@ def initialize(setup_manager, scope, studio, name, run_opts)
       @studio = studio
       @name = name
       @kit = setup_manager.kit
-      @pool = 'Default Pool'
       @logger.info("Starting client #{name}")
       @runner = setup_manager.run_client(scope, @name, run_opts)
       scope << self
     end
 
+    def pool
+      @studio.list_pools.each do |pool|
+        return pool['name'] if pool['machines'].any? { |machine| machine['name'].eql?(@name) }
+      end
+
+      'Default Pool'
```

Comment:
It shouldn't return `'Default Pool'` when `list_pools` says it's not in `Default Pool`.